src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -418,6 +424,48 @@ public unsafe void SetResponseHeaders()
             }
         }
 
+        public unsafe void SetResponseTrailers()
+        {
+            HttpResponseTrailers.IsReadOnly = true;
+            foreach (var headerPair in HttpResponseTrailers)
+            {
+                var headerValues = headerPair.Value;
+
+                if (headerValues.Count == 0)
+                {
+                    continue;
+                }
+
+                string headerValue;
+                if (headerValues.Count == 1)
+                {
+                    headerValue = headerValues[0];
+                }
+                else
+                {
+                    var builder = new StringBuilder();
```

Comment:
Why aren't you using ToString on the StringValues?

Reply:
Api doesn't allow multiple headers with the same name. The api for HttpResponseSetTrailer doesn't have an fReplace flag like HttpResponseSetHeader. I chatted with Levi about it in the past, and at the time we agreed not to add it.

Reply:
Actually this may be an issue. Looking into it.

Reply:
Yeah, I don't think there is a way way to set multiple trailers with the same name right now. We may need to add the fReplace to the API.

@PranavSenthilnathan, currently the issue is if you would like to set two response trailers with the same name, ex:

MyHeaderName: Value1
MyHeaderName: Value2

You can't currently do that today.

@Tratcher can you help evaluate how big of a deal this is? Currently ResponseTrailers_MultipleValues_SentAsSeparateHeaders will fail because the headers aren't in different value and rather the same:

```
    Assert.Equal() Failure
    Expected: String[] ["TrailerValue0", "TrailerValue1"]
    Actual:   String[] ["TrailerValue0,TrailerValue1"]
```

Reply:
Officially the two formats are equivalent. In practice there are some select headers and clients that require multiple headers (e.g. Set-Cookie). If we have the chance to support multiple headers now we should do it.

Reply:
@pgovind could we add this? It should be effectively the same implementation that is there for headers, where you have a fReplace flag.

Reply:
@jkotalik Do you recall what the reasoning was for not adding this previously? Was it just that we didn't think it would be needed?

Reply:
Yeah we thought it wouldn't be needed. Originally, I thought that you could still write multiple headers with the same name, but that we needed to handle it in managed code. However, I don't believe that is the case.

Having a header format like:
MyHeaderName: Value1,Value2

should be equivalent, however there are some headers that require the headers separated.

I looked at the code in IIS and I don't think it's hard to add this flag. Should be the same code that's present for the set_header path.

Reply:
Working on this now. Will update folks once I'm done.